### PR TITLE
Feature/test user flows writing

### PR DESCRIPTION
**Pull request by Alynn022**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (change which fixes an issue)
- [x] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

 
- Add user flow tests with cypress 

 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# Where should reviewers start?
UserFlow_spec.tsx

# How Has This Been Tested?

Please check the option(s) that best describe your tests. 

- [x] New spec files were added
- [ ] Tests were added to existing spec files
- [ ] Some tests were changed
- [ ] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Please add further details of your tests here.

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [ ] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: cypress/integration/UserFlow_spec.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    .get('.continue-writing-button').click()
+    .get('.modal').should('not.exist')
+  })
+  it('should have a pop up a modal when time is up', () => {
+    cy.get('.choose-time-btn').should('exist').click()
+    .get('select')
+    .select('60')
+    .get('p').should('have.text', 'target: 01:00')
+    .get('.start-stop-btn').click()
+    .wait(60000)
```
Comment (philalewis) [inline cypress/integration/UserFlow_spec.tsx:52]: This is good to know.

File: cypress/integration/UserFlow_spec.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  })
+
+  it('should be able to start writing in the text area', () => {
+    cy.get('.choose-time-btn').should('exist').click()
+    .get('select')
+    .select('60')
+    .get('p').should('have.text', 'target: 01:00')
+    .get('.start-stop-btn').click()
+    .get('textarea').type('Our story begins....')
+    .get('textarea').should('have.text', 'Our story begins....')
```
Comment (philalewis) [inline cypress/integration/UserFlow_spec.tsx:79]: We could probably combine this test with the last one if we wanted.

File: cypress/integration/UserFlow_spec.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    .get('p').should('have.text', 'target: 01:00')
+    .get('.start-stop-btn').click()
+    .get('input').type('One minute writer')
+    .get('input').should('have.value', 'One minute writer')
+    .get('textarea').type('Our story begins....')
+    .get('textarea').should('have.text', 'Our story begins....')
+    .get('.start-stop-btn').click()
+    .get('.save-writing-button').click()
+    .intercept('GET', 'https://one-minute-writer-be.herokuapp.com/graphql')
+    .visit('https://one-minute-writer.herokuapp.com')
```
Comment (philalewis) [inline cypress/integration/UserFlow_spec.tsx:95]: Should we see if this was added to the stories? I think we might have to make another intercept for the get request on dashboard page load.

**Review by philalewis — APPROVED** [review]
We should add some sad path testing, such as if the user doens't enter anything into the title input or the text input.

File: cypress/integration/UserFlow_spec.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    cy.visit('https://one-minute-writer.herokuapp.com/writing-page')
+  })
+
+  it('should be able to get new inspiration word', () => {
+    cy.get('.word-inspo').should('exist')
+    // .get('.new-word-btn').click()
+    // .get('.word-inspo').should('exist')
+})
+
+  // it('should be able to get new inspiration sound', () => {
```
Comment (Jayquintana) [inline cypress/integration/UserFlow_spec.tsx:31]: if not using remember to delete later

**Review by Jayquintana — APPROVED** [review]

**Pull request merged**